test/features/georoutingv2/georoutingv2.test.js

```diff
@@ -0,0 +1,470 @@
+    expect(modal).to.be.null;
+    // cleanup
+    setUserCountryFromIP('CH');
+  });
+
+  it('Will not display non found link if fallback routing is off', async () => {
+    // prepare
+    stubFallbackMetadata('off');
+    stubHeadRequestToReturnVal('/ch_it', false);
+    await init(mockConfig, createTag, getMetadata, loadBlock, loadStyle);
+    const modal = document.querySelector('.dialog-modal');
+    // assert
+    const tabs = modal.querySelectorAll('.tabpanel');
+    expect(tabs.length).to.be.equal(2);
+    const germanTab = tabs[0];
+    const frenchTab = tabs[1];
+    const germanData = mockGeoroutingJson.georouting.data.find((d) => d.prefix === 'ch_de');
+    const frenchData = mockGeoroutingJson.georouting.data.find((d) => d.prefix === 'ch_fr');
+    const usData = mockGeoroutingJson.georouting.data.find((d) => d.prefix === '');
+    expect(germanTab.querySelector('h3').textContent).to.be.equal('Sie besuchen Adobe.com für die Vereinigten Staaten.');
+    expect(germanTab.querySelector('p').textContent).to.be.equal(germanData.text);
+    expect(germanTab.querySelector('.con-button').textContent).to.be.equal(germanData.button);
+    expect(germanTab.querySelectorAll('a')[1].textContent).to.be.equal(usData.button);
+
+    expect(frenchTab.querySelector('h3').textContent).to.be.equal('Vous visitez Adobe.com pour les Etats-Unis.');
+    expect(frenchTab.querySelector('p').textContent).to.be.equal(frenchData.text);
+    expect(frenchTab.querySelector('.con-button').textContent).to.be.equal(frenchData.button);
+    expect(frenchTab.querySelectorAll('a')[1].textContent).to.be.equal(usData.button);
+    // cleanup
+    stubFallbackMetadata('on');
+    stubHeadRequestToReturnVal('/ch_it', true);
+  });
+
+  it('Will show fallback links if fallbackrouting is on and page exist request fails', async () => {
+    // prepare
+    stubHeadRequestToReturnVal('/ch_it', false);
+    await init(mockConfig, createTag, getMetadata, loadBlock, loadStyle);
+    const modal = document.querySelector('.dialog-modal');
+    // assert
+    expect(modal).to.not.be.null;
+    const tabs = modal.querySelectorAll('.tabpanel');
+    expect(tabs.length).to.be.equal(3);
+    // cleanup
+    stubHeadRequestToReturnVal('/ch_it', true);
+  });
+
+  it('Will not display modal if no single link was found and if fallback routing is off', async () => {
+    // prepare
+    stubFallbackMetadata('off');
+    stubHeadRequestToReturnVal('/ch_de', false);
+    stubHeadRequestToReturnVal('/ch_it', false);
+    stubHeadRequestToReturnVal('/ch_fr', false);
+    await init(mockConfig, createTag, getMetadata, loadBlock, loadStyle);
+    const modal = document.querySelector('.dialog-modal');
+    // assert
+    expect(modal).to.be.null;
+    // cleanup
+    stubFallbackMetadata('on');
+    stubHeadRequestToReturnVal('/ch_de', true);
+    stubHeadRequestToReturnVal('/ch_it', true);
+    stubHeadRequestToReturnVal('/ch_fr', true);
+  });
+
+  it('Closes picker properly', async () => {
```

Comment:
This test seems to test for opening the picker, checking the content, and then closing the picker. Could breakout if desired, or change the string. Personally my preference is to break it out. Could be a candidate for more `describe` blocks.

Reply:
changed the string 👍